src/attribution/onINP.ts

```diff
@@ -25,38 +28,224 @@ import {
+ * entries to an `entryToRenderTimeMap` WeakMap so that the "grouped" render
+ * times can be looked up later.
+ */
+const groupEntriesByRenderTime = (entries: PerformanceEventTiming[]) => {
+  entries.forEach((entry) => {
+    let renderTime = entry.startTime + entry.duration;
+    let previousRenderTime;
+
+    // Iterate of all previous render times in reverse order to find a match.
+    // Go in reverse since the most likely match will be at the end.
+    for (let i = previousRenderTimes.length - 1; i >= 0; i--) {
+      previousRenderTime = previousRenderTimes[i];
+
+      // If a previous render time is within 8ms of the current render time,
+      // assume they were part of the same frame and re-use the previous time.
+      // Also break out of the loop because all subsequent times will be newer.
+      if (Math.abs(renderTime - previousRenderTime) <= 8) {
+        const group = pendingEntriesGroupMap.get(previousRenderTime)!;
+        group.startTime = Math.min(entry.startTime, group.startTime);
+        group.processingStart = Math.min(
+          entry.processingStart,
+          group.processingStart,
+        );
+        group.processingEnd = Math.max(
+          entry.processingEnd,
+          group.processingEnd,
+        );
+        group.entries.push(entry);
+
+        renderTime = previousRenderTime;
+        break;
+      }
+    }
+
+    // If there was no matching render time, assume this is a new frame.
+    if (renderTime !== previousRenderTime) {
+      previousRenderTimes.push(renderTime);
+      pendingEntriesGroupMap.set(renderTime, {
+        startTime: entry.startTime,
+        processingStart: entry.processingStart,
+        processingEnd: entry.processingEnd,
+        entries: [entry],
+      });
+    }
+
+    // Store the grouped render time for this entry for reference later.
+    if (entry.interactionId || entry.entryType === 'first-input') {
+      entryToRenderTimeMap.set(entry, renderTime);
+    }
+  });
+
+  // Queue cleanup of entries that are not part of any INP candidates.
+  if (idleCallback < 0) {
+    idleCallback = requestIdleCallback(cleanupEntries);
+  }
+};
+
+const cleanupEntries = () => {
+  // The list of previous render times is used to handle cases where
+  // events are dispatched out of order. When this happens they're generally
+  // only off by a frame or two, so keeping the most recent 10 should be
+  // more than sufficient.
+  previousRenderTimes = previousRenderTimes.slice(-10);
+
+  // Keep all render times that are part of a pending INP candidate or
+  // that occurred within 2 seconds of the most recent render time.
+  const renderTimesToKeep = new Set(
+    (previousRenderTimes as (number | undefined)[]).concat(
+      longestInteractionList.map((i) => entryToRenderTimeMap.get(i.entries[0])),
+    ),
+  );
+
+  pendingEntriesGroupMap.forEach((_, key) => {
+    renderTimesToKeep.has(key) || pendingEntriesGroupMap.delete(key);
+  });
```

Comment:
Is the 2 second check still here or has that been removed now?

Reply:
It's been removed in favor of what's explain in the comments above ([line 144-148](https://github.com/GoogleChrome/web-vitals/blob/a41d463664913ab734f0b6685bf4b57777df2f45/src/attribution/onINP.ts#L144-L148)).

Reply:
So can we remove it from the comment?

Reply:
Oops! (Good catch.)